backend/tracim_backend/lib/search/elasticsearch_search/es_analyzers.py

```diff
@@ -0,0 +1,125 @@
+
+# INFO - G.M - 2022-04-08 - Elasticsearch "pipeline" for analyser is :
+# - char_filter : modify the entry
+# - tokenizer : split the entry into token
+# - filter/token_filter : modify the token
+# As existing tokenizer doesn't look to be extendable with new separator, the workaround used here to
+# be able to split on "_" is to replace them in char_filter to "-" which is an existing separator.
+underscore_as_minus = analysis.char_filter("underscore_as_space", type="mapping", mappings=["_=>-"])
+
+# INFO - G.M - 2022-04-07 - Locally  Testing analyser without running tracim is possible with elyzer
+# (need same name python package)  with command like :
+#  elyzer --es "http://localhost:9200" --index my-tracim-content --analyzer folding "Mary had_a little lamb"
+# This permit to better understand each step of the analyzer process.
+
+# INFO - G.M - 2019-05-23 - search_analyser: do search for content given an some similar word
+folding = analyzer(
+    "folding",
+    tokenizer="standard",
+    filter=["lowercase", "asciifolding"],
+    char_filter=underscore_as_minus,
+)
+# INFO - G.M - 2019-05-23 -  index_analysers, index edge ngram for autocompletion and strip html for indexing
+edge_ngram_token_filter = analysis.token_filter(
+    "edge_ngram_filter", type="edge_ngram", min_ngram=2, max_gram=20
+)
+# NOTE 2021-11-02 - S.G. - Configuring a maximum length for tokens (e.g. words)
+# so that the following error does not occur:
+# https://stackoverflow.com/questions/24019868/utf8-encoding-is-longer-than-the-max-length-32766
+# It can happen in HTML custom properties in which images or videos are embedded
+# when the custom properties schema does not set '"format"="html"'.
+max_token_length_filter = analysis.token_filter(
+    "max_token_length_filter", type="length", min=0, max=32766
+)
+
+edge_ngram_folding = analyzer(
+    "edge_ngram_folding",
+    tokenizer="standard",
+    filter=[max_token_length_filter, "lowercase", "asciifolding", edge_ngram_token_filter],
+    char_filter=underscore_as_minus,
+)
+html_folding = analyzer(
+    "html_folding",
+    tokenizer="standard",
+    filter=[max_token_length_filter, "lowercase", "asciifolding", edge_ngram_token_filter],
+    char_filter=["html_strip", underscore_as_minus],
+)
+html_exact_folding = analyzer(
+    "html_exact_folding",
+    tokenizer="standard",
+    filter=[max_token_length_filter],
+    char_filter=["html_strip", underscore_as_minus],
+)
+
+# NOTE - GM - 2022-04-20 : Custom Language analyzer based on reference lang analyzer:
+# custom behavior include:
+# - split on "_" with underscore_as_minus char filter.
+# about original lang analyzer, see:
+# https://www.elastic.co/guide/en/elasticsearch/reference/7.13/analysis-lang-analyzer.html
+
+#  French  #
+french_elision = analysis.token_filter(
+    "french_elision",
+    type="elision",
+    article_case=True,
+    articles=["l", "m", "t", "qu", "n", "s", "j", "d", "c", "jusqu", "quoiqu", "lorsqu", "puisqu"],
+)
```

Comment:
I'm not sure to understand what is the point of the french_elision here
If it's set for french, why not for other language?
Also, I guess the ``s`` in french is for ``s'est`` but it could also be for ``it's`` in english?

Reply:
The point is just to "copy/paste" the default language analyzer here:  https://www.elastic.co/guide/en/elasticsearch/reference/7.13/analysis-lang-analyzer.html . Only the french one as elision. We may improve the analyzer per lang later.